app/controllers/planning_applications/neighbour_letters_controller.rb

```diff
@@ -113,8 +115,16 @@ def require_reason_when_resending
       render :index and return
     end
 
+    def require_deadline
+      return unless consultation_params[:deadline].blank?
+      flash.now[:alert] = "Deadline is required"
+      render :index and return
+    end
+
     def update_consultation!
-      @consultation.update!(consultation_params.except(:resend_existing, :resend_reason).merge(status: "in_progress"))
+      @consultation.update!(consultation_params.except(:resend_existing, :resend_reason, :deadline).merge(status: "in_progress"))
+
+      @consultation.extend_deadline(consultation_params[:deadline].to_i.days.from_now.end_of_day)
```

Comment:
Can we not push this onto the consultation model and use a custom validation context like we do for consultees with a virtual attribute

Reply:
By virtual attribute do you mean a `deadline=` method on Consultation and letting it be invoked through the `update!`? If so I can give that a go, yes.

Reply:
I mean using the Active Record Attributes API to define custom attributes on the model like we do here:

https://github.com/unboxed/bops/blob/df4af49c386f7384b9439060b2e6a229b1c95176/app/models/consultation.rb#L18-L21

Reply:
This is then used with a custom validation context (e.g `on: :send_consultee_emails`):

https://github.com/unboxed/bops/blob/df4af49c386f7384b9439060b2e6a229b1c95176/app/models/consultation.rb#L42-L50

It means you can use the standard form error message handling